controllers/gce/loadbalancers/loadbalancers.go

```diff
@@ -384,32 +408,28 @@ func (l *L7) checkSSLCert() (err error) {
 	// TODO: Clean this code up into a ring buffer.
 	primaryCertName := l.namer.Truncate(fmt.Sprintf("%v-%v", sslCertPrefix, l.Name))
 	secondaryCertName := l.namer.Truncate(fmt.Sprintf("%v-%d-%v", sslCertPrefix, 1, l.Name))
-	certName = primaryCertName
-	if l.sslCert != nil {
-		certName = l.sslCert.Name
-	}
-
-	// Skip error checking because error-ing out will retry and loop, when we
-	// should create/update the cert if there is an error or does not exist.
-	cert, _ := l.cloud.GetSslCertificate(certName)
 
 	// PrivateKey is write only, so compare certs alone. We're assuming that
 	// no one will change just the key. We can remember the key and compare,
 	// but a bug could end up leaking it, which feels worse.
-	if cert == nil || ingCert != cert.Certificate {
-
-		certChanged := cert != nil && (ingCert != cert.Certificate)
-		if certChanged {
-			if certName == primaryCertName {
-				certName = secondaryCertName
-			} else {
-				certName = primaryCertName
+	if l.sslCert == nil || ingCert != l.sslCert.Certificate {
+		newCertName := primaryCertName
+		if l.sslCert != nil && (ingCert != l.sslCert.Certificate) {
+			if l.sslCert.Name == primaryCertName {
+				newCertName = secondaryCertName
 			}
 		}
 
-		glog.Infof("Creating new sslCertificates %v for %v", certName, l.Name)
-		cert, err = l.cloud.CreateSslCertificate(&compute.SslCertificate{
-			Name:        certName,
+		// Perform a delete in case a certificate exists with the exact name
+		// This certificate should be unused since we check the target proxy's certificate prior
+		// to this point. Although, it's possible an actor pointed a target proxy to this certificate.
+		if err = l.cloud.DeleteSslCertificate(newCertName); err != nil && !utils.IsHTTPErrorCode(err, http.StatusNotFound) {
```

Comment:
this is a little tangential to thie review, but a lot of the StatusNotFound if checks can be made more succinct by 

```
func ignoreHTTPNotFound(err error) error {
  if err != nil && utils.IsHTTPErrorCode(err, http.StatusNotFound) { return nil }
  return err
}
```

then all of these sites will be 
```
if err = ignoreHTTPNotFound(l.cloud.DeleteSslCertificate(newCertName)); err != nil {
}
```